File: src/Entity/Restaurant.php
<?php

namespace App\Entity;

use App\Repository\RestaurantRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
use Symfony\Component\Serializer\Annotation\Groups ;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=RestaurantRepository::class)
 */
class Restaurant
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("post:read")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Length(
     *      min = 2,
     *      max = 15,
     *      minMessage = "nom Restaurant doit etre au minimum {{ limit }} characters long",
     *      maxMessage = "nom Restaurant ne doit pas passer {{ limit }} characters")
     * @Assert\NotNull
     * @Groups("post:read")
     *
     */
    private $nomRestaurant;



    /**
     * @ORM\Column(type="blob")
     * @Groups("restaurant")
     */
    private $imageRestaurant;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("post:read")
     */
    private $nomImage;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("post:read")
     */
    private $categorieRestaurant;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\Length(
     *      min = 5,
     *      max = 20,
     *      minMessage = "adresse doit etre au minimum {{ limit }} characters long",
     *      maxMessage = "adresse ne doit pas depasser {{ limit }} characters"
     * )
     * @Assert\NotNull
     * @Groups("post:read")
     *
     */
    private $addresse;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("post:read")
     *
     */
    private $cite;

    /**
     * @ORM\Column(type="integer")
     * @Assert\Range(
     *      min = 1000,
     *      max = 9999,
     *      notInRangeMessage = "code postal doit etre en 4 chiffres")
     * @Groups("post:read")
     */
    private $codePostal;

    /**
     * @ORM\Column(type="datetime")
     * @Groups("post:read")
     */
    private $createdAt;

    /**
     * @ORM\Column(type="time")
     * @Groups("post:read")
     *
     *
     */
    private $heureOuverture;

    /**
     * @ORM\Column(type="time")
     * @Assert\GreaterThan(propertyPath="heureOuverture")
     * @Groups("post:read")
     *
     */
    private $heureFermeture;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("post:read")
     */
    private $status;

    /**
     * @ORM\Column(type="text")
     * @Assert\Length(
     *      min = 10,
     *      max = 200,
     *      minMessage = "description doit etre au minimum {{ limit }} characters long",
     *      maxMessage = "description ne doit pas depasser {{ limit }} characters"
     * )
     * @Assert\NotNull
     * @Groups("post:read")
     *
     */
    private $description;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="restaurants")
     * @Groups("post:read")
     */
    private $user;

    /**
     * @ORM\OneToMany(targetEntity=Produits::class, mappedBy="restaurant", orphanRemoval=true)
     * @Groups("post:produits")
     */
    private $produits;

    /**
     * @ORM\OneToMany(targetEntity=Reclamation::class, mappedBy="restaurant", orphanRemoval=true)
     * @Groups("post:read")
     */
    private $reclamations;

    protected $captchaCode;



    /**
     * @ORM\OneToMany(targetEntity=Evenement::class, mappedBy="Restaurant")
     */
    private $evenements;




    public function __construct()
    {
        $this->produits = new ArrayCollection();
        $this->reclamations = new ArrayCollection();

        $this->evenements = new ArrayCollection();

    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomRestaurant(): ?string
    {
        return $this->nomRestaurant;
    }

    public function setNomRestaurant(string $nomRestaurant): self
    {
        $this->nomRestaurant = $nomRestaurant;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getImageRestaurant()
    {
        return $this->imageRestaurant;
    }

    public function setImageRestaurant($imageRestaurant): self
    {
        $this->imageRestaurant = $imageRestaurant;

        return $this;
    }

    public function getNomImage(): ?string
    {
        return $this->nomImage;
    }

    public function setNomImage(string $nomImage): self
    {
        $this->nomImage = $nomImage;

        return $this;
    }

    public function getCategorieRestaurant(): ?string
    {
        return $this->categorieRestaurant;
    }

    public function setCategorieRestaurant(string $categorieRestaurant): self
    {
        $this->categorieRestaurant = $categorieRestaurant;

        return $this;
    }

    public function getAddresse(): ?string
    {
        return $this->addresse;
    }

    public function setAddresse(string $addresse): self
    {
        $this->addresse = $addresse;

        return $this;
    }

    public function getCite(): ?string
    {
        return $this->cite;
    }

    public function setCite(string $cite): self
    {
        $this->cite = $cite;

        return $this;
    }

    public function getCodePostal(): ?int
    {
        return $this->codePostal;
    }

    public function setCodePostal(int $codePostal): self
    {
        $this->codePostal = $codePostal;

        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function setCreatedAt(\DateTimeInterface $createdAt): self
    {
        $this->createdAt = $createdAt;

        return $this;
    }

    public function getHeureOuverture(): ?\DateTimeInterface
    {
        return $this->heureOuverture;
    }

    public function setHeureOuverture(\DateTimeInterface $heureOuverture): self
    {
        $this->heureOuverture = $heureOuverture;

        return $this;
    }

    public function getHeureFermeture(): ?\DateTimeInterface
    {
        return $this->heureFermeture;
    }

    public function setHeureFermeture(\DateTimeInterface $heureFermeture): self
    {
        $this->heureFermeture = $heureFermeture;

        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    /**
     * @return Collection<int, Produits>
     */
    public function getProduits(): Collection
    {
        return $this->produits;
    }

    public function addProduit(Produits $produit): self
    {
        if (!$this->produits->contains($produit)) {
            $this->produits[] = $produit;
            $produit->setRestaurant($this);
        }

        return $this;
    }

    public function removeProduit(Produits $produit): self
    {
        if ($this->produits->removeElement($produit)) {
            // set the owning side to null (unless already changed)
            if ($produit->getRestaurant() === $this) {
                $produit->setRestaurant(null);
            }
        }

        return $this;
    }

    /**
     * @return Collection<int, Reclamation>
     */
    public function getReclamations(): Collection
    {
        return $this->reclamations;
    }

    public function addReclamation(Reclamation $reclamation): self
    {
        if (!$this->reclamations->contains($reclamation)) {
            $this->reclamations[] = $reclamation;
            $reclamation->setRestaurant($this);
        }

        return $this;
    }

    public function removeReclamation(Reclamation $reclamation): self
    {
        if ($this->reclamations->removeElement($reclamation)) {
            // set the owning side to null (unless already changed)
            if ($reclamation->getRestaurant() === $this) {
                $reclamation->setRestaurant(null);
            }
        }

        return $this;
    }
    public function getCaptchaCode()
    {
        return$this->captchaCode;
    }

    /**
     * @param mixed $captchaCode
     */
    public function setCaptchaCode($captchaCode): void
    {
        $this->captchaCode = $captchaCode;
    }



    /**
     * @return Collection<int, Evenement>
     */
    public function getEvenements(): Collection
    {
        return $this->evenements;
    }

    public function addEvenement(Evenement $evenement): self
    {
        if (!$this->evenements->contains($evenement)) {
            $this->evenements[] = $evenement;
            $evenement->setRestaurant($this);
        }

        return $this;
    }

    public function removeEvenement(Evenement $evenement): self
    {
        if ($this->evenements->removeElement($evenement)) {
            // set the owning side to null (unless already changed)
            if ($evenement->getRestaurant() === $this) {
                $evenement->setRestaurant(null);
            }
        }

        return $this;
    }






}

File: src/com/mycompagny/entities/Restaurant.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompagny.entities;

import java.util.Date;

/**
 *
 * @author haithem
 */
public class Restaurant {

    private int id;
    private String nom,description,addresse,cite,heureOuverture,categorieRestaurant,heureFermeture,gouvernorat;
    private int codePostal;
    private String createdAt;
    private String imageRestaurant;
    private String nomImage;
    private String status;
    private int idUser;

    public Restaurant(){

    }


    public Restaurant(int id, String nom, String description, String addresse, String cite, String heureOuverture, String categorieRestaurant, String heureFermeture, String gouvernorat, int codePostal, String createdAt, String imageRestaurant, String nomImage, String status) {
        this.id = id;
        this.nom = nom;
        this.description = description;
        this.addresse = addresse;
        this.cite = cite;
        this.heureOuverture = heureOuverture;
        this.categorieRestaurant = categorieRestaurant;
        this.heureFermeture = heureFermeture;
        this.gouvernorat = gouvernorat;
        this.codePostal = codePostal;
        this.createdAt = createdAt;
        this.imageRestaurant = imageRestaurant;
        this.nomImage = nomImage;
        this.status = status;
    }

    public Restaurant(int idUser, String nom, String categorieRestaurant, String description, String gouvernorat, String addresse, String cite, int codePostal, String heureOuverture, String heureFermeture, String imageRestaurant, String nomImage, String status, String createdAt) {
        this.nom = nom;
        this.description = description;
        this.createdAt = createdAt;
        this.addresse = addresse;
        this.cite = cite;
        this.codePostal = codePostal;
        this.heureOuverture = heureOuverture;
        this.categorieRestaurant = categorieRestaurant;
        this.heureFermeture = heureFermeture ;
        this.gouvernorat = gouvernorat ;
        this.imageRestaurant = imageRestaurant;
        this.nomImage = nomImage;
        this.status= status;
        this.idUser= idUser;


    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }



    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }
    public String getAddresse() {
        return addresse;
    }
    public void setAddresse(String addresse) {
        this.addresse = addresse;
    }
    public String getCite() {
        return cite;
    }
    public void setCite(String cite) {
        this.cite = cite;
    }
    public int getCodePostal() {
        return codePostal;
    }
    public void setCodePostal(int codePostal) {
        this.codePostal = codePostal;
    }
    public String getHeureOuverture() {
        return heureOuverture;
    }
    public void setHeureOuverture(String heureOuverture) {
        this.heureOuverture = heureOuverture;
    }
    public String getCategorieRestaurant() {
        return categorieRestaurant;
    }
    public void setCategorieRestaurant(String categorieRestaurant) {
        this.categorieRestaurant = categorieRestaurant;
    }

    public String getHeureFermeture() {
        return heureFermeture;
    }

    public void setHeureFermeture(String heureFermeture) {
        this.heureFermeture = heureFermeture;
    }

    public String getGouvernorat() {
        return gouvernorat;
    }

    public void setGouvernorat(String gouvernorat) {
        this.gouvernorat = gouvernorat;
    }

    public String getImageRestaurant() {
        return imageRestaurant;
    }

    public void setImageRestaurant(String imageRestaurant) {
        this.imageRestaurant = imageRestaurant;
    }

    public String getNomImage() {
        return nomImage;
    }

    public void setNomImage(String nomImage) {
        this.nomImage = nomImage;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}
